src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/DeleteUserProfileIntTest.java

```diff
@@ -0,0 +1,154 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.userprofileapi.controller.request.UserProfileDataRequest;
+import uk.gov.hmcts.reform.userprofileapi.controller.response.UserProfileCreationResponse;
+import uk.gov.hmcts.reform.userprofileapi.controller.response.UserProfilesDeletionResponse;
+import uk.gov.hmcts.reform.userprofileapi.domain.entities.Audit;
+import uk.gov.hmcts.reform.userprofileapi.domain.entities.UserProfile;
+import uk.gov.hmcts.reform.userprofileapi.domain.enums.ResponseSource;
+import uk.gov.hmcts.reform.userprofileapi.resource.UserProfileCreationData;
+
+@RunWith(SpringIntegrationSerenityRunner.class)
+@SpringBootTest(webEnvironment = MOCK)
+@Transactional
+public class DeleteUserProfileIntTest extends AuthorizationEnabledIntegrationTest {
+
+    @Before
+    public void setUp() {
+        this.mockMvc = webAppContextSetup(webApplicationContext).build();
+    }
+
+    @Test
+    public void should_return_204_and_delete_user_profile_resource() throws Exception {
+
+        UserProfileCreationData data = buildCreateUserProfileData();
+        UserProfileCreationResponse createdResource =
+            userProfileRequestHandlerTest.sendPost(
+                mockMvc,
+                APP_BASE_PATH,
+                data,
+                CREATED,
+                UserProfileCreationResponse.class
+            );
+
+        verifyUserProfileCreation(createdResource, CREATED, data);
+        List<String> userIds = new ArrayList<String>();
+        userIds.add(createdResource.getIdamId());
+        UserProfileDataRequest deletionRequest = buildUserProfileDataRequest(userIds);
+        userProfileRequestHandlerTest.sendDelete(mockMvc,
+                APP_BASE_PATH,
+                deletionRequest,
+                NO_CONTENT,
+                UserProfilesDeletionResponse.class);
+
+        verifyUserProfileDeletion();
+
+    }
+
+    @Test
+    public void should_return_204_and_delete_multiple_user_profile_resource() throws Exception {
+
+        UserProfileCreationData data = buildCreateUserProfileData();
+        //user profile one created
+        UserProfileCreationResponse createdResourceOne =
+                userProfileRequestHandlerTest.sendPost(
+                        mockMvc,
+                        APP_BASE_PATH,
+                        data,
+                        CREATED,
+                        UserProfileCreationResponse.class
+                );
+
+        verifyUserProfileCreation(createdResourceOne, CREATED, data);
+        UserProfileCreationData data2 = buildCreateUserProfileData();
+        //user profile two created
+        UserProfileCreationResponse createdResourceTwo =
+                userProfileRequestHandlerTest.sendPost(
+                        mockMvc,
+                        APP_BASE_PATH,
+                        data2,
+                        CREATED,
```

Comment:
can we add scenario for multiple delete request - where 1 out of all users is not found or delete not successful?

Reply:
ok. added test case for that